### PR TITLE
JAMES-2558 Tests should rely on SpamAssassin 3.4.2

### PR DESCRIPTION
**Pull request by chibenwa**
Fixes CVE-2017-15705, CVE-2016-1238, CVE-2018-11780 & CVE-2018-11781

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not build image: The command '/bin/sh -c apt-get update &&     DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y         gpg         libio-socket-ip-perl         libmail-dkim-perl         libnet-ident-perl         libsocket-getaddrinfo-perl         pyzor         razor         libdbi-perl         libdbd-pg-perl         spamassassin=${SPAMASSASSIN_VERSION}* &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
```

Packaging not yet available exception?

**Comment by chibenwa** [comment]
The container inherit Debian Stretch from the postgres container https://github.com/docker-library/postgres/blob/3f585c58df93e93b730c09a13e8904b96fa20c58/10/Dockerfile

But stretch is only 3.4.1 based cf https://packages.debian.org/fr/spamassassin

Switching to testing or unstable do not seem wize.

Maybe we should curl + dpkg some trusted 3.4.2 package over the internet (or update our APT lists)

**Comment by chibenwa** [comment]
This should be done in the following days : https://tracker.debian.org/pkg/spamassassin

**Comment by chibenwa** [comment]
This should be done in the following days : https://tracker.debian.org/pkg/spamassassin

We can monitor progress here...

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR] Errors: 
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldAllowToForgetSpam » ContainerFetch Can...
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldReturnTrueWhenLearningWorks » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsHamWhenMessageAlreadyLearnedAsHam » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenKnownAsSpam » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenMessageAlreadyLearnedAsSpam » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHeadersField » ContainerFetch Can'...
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHitsField » ContainerFetch Can't g...
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyRequiredHitsField » ContainerFetch
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [INFO] 
[fb95fe1e714c712cc6e40ea95a28138aa037bff4] [ERROR] Tests run: 14, Failures: 0, Errors: 9, Skipped: 0
```

**Comment by chibenwa** [comment]
Yes no need to relaunch it, the debian package is not yet available

**Comment by chibenwa** [comment]
test this please

(just to have a failing build report instead of an inapropriate pending build report, packaging is still not yet available...)

**Comment by trantienduchn** [comment]
here you are
```
[null] Tests in error: 
[null]   SpamAssassinInvokerTest.learnAsHamShouldAllowToForgetSpam » ContainerFetch Can...
[null]   SpamAssassinInvokerTest.learnAsHamShouldReturnTrueWhenLearningWorks » ContainerFetch
[null]   SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks » ContainerFetch
[null]   SpamAssassinInvokerTest.scanMailShouldMarkAsHamWhenMessageAlreadyLearnedAsHam » ContainerFetch
[null]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenKnownAsSpam » ContainerFetch
[null]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenMessageAlreadyLearnedAsSpam » ContainerFetch
[null]   SpamAssassinInvokerTest.scanMailShouldModifyHeadersField » ContainerFetch Can'...
[null]   SpamAssassinInvokerTest.scanMailShouldModifyHitsField » ContainerFetch Can't g...
[null]   SpamAssassinInvokerTest.scanMailShouldModifyRequiredHitsField » ContainerFetch
[null] 
[null] Tests run: 111, Failures: 0, Errors: 9, Skipped: 0
```

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldAllowToForgetSpam » ContainerFetch Can...
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldReturnTrueWhenLearningWorks » ContainerFetch
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks » ContainerFetch
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsHamWhenMessageAlreadyLearnedAsHam » ContainerFetch
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenKnownAsSpam » ContainerFetch
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenMessageAlreadyLearnedAsSpam » ContainerFetch
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHeadersField » ContainerFetch Can'...
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHitsField » ContainerFetch Can't g...
[ba6fa212f3e3857376b43b3714c65dfd32cd063c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyRequiredHitsField » ContainerFetch
```

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
still
```
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR] Errors: 
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldAllowToForgetSpam » ContainerLaunch Co...
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldReturnTrueWhenLearningWorks » ContainerLaunch
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks » ContainerLaunch
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsHamWhenMessageAlreadyLearnedAsHam » ContainerLaunch
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenKnownAsSpam » ContainerLaunch
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenMessageAlreadyLearnedAsSpam » ContainerLaunch
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHeadersField » ContainerLaunch Con...
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHitsField » ContainerLaunch Contai...
[e244bf048151fffb286567264e4f6e7a256a899b] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyRequiredHitsField » ContainerLaunch
```

**Comment by rouazana** [comment]
still
```
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [INFO] Running org.apache.james.spamassassin.SpamAssassinResultTest
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.879 s - in org.apache.james.spamassassin.SpamAssassinResultTest
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [INFO] 
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [INFO] Results:
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [INFO] 
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR] Errors: 
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldAllowToForgetSpam » ContainerLaunch Co...
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.learnAsHamShouldReturnTrueWhenLearningWorks » ContainerLaunch
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.learnAsSpamShouldReturnTrueWhenLearningWorks » ContainerLaunch
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsHamWhenMessageAlreadyLearnedAsHam » ContainerLaunch
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenKnownAsSpam » ContainerLaunch
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldMarkAsSpamWhenMessageAlreadyLearnedAsSpam » ContainerLaunch
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHeadersField » ContainerLaunch Con...
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyHitsField » ContainerLaunch Contai...
[ba593118a7ae3798efb3757c7ee8b4c7d22cca6c] [ERROR]   SpamAssassinInvokerTest.scanMailShouldModifyRequiredHitsField » ContainerLaunch
```
test this please...

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java [hunk 1]
```diff
@@ -51,8 +53,9 @@ public SpamAssassinExtension() {
                 .withFileFromClasspath("run.sh", "docker/spamassassin/run.sh")
                 .withFileFromClasspath("spamd.sh", "docker/spamassassin/spamd.sh")
                 .withFileFromClasspath("rule-update.sh", "docker/spamassassin/rule-update.sh")
-                .withFileFromClasspath("bayes_pg.sql", "docker/spamassassin/bayes_pg.sql"));
-        spamAssassinContainer.waitingFor(new SpamAssassinWaitStrategy(spamAssassinContainer));
+                .withFileFromClasspath("bayes_pg.sql", "docker/spamassassin/bayes_pg.sql"))
+            .withStartupTimeout(STARTUP_TIMEOUT);
```
Comment (mbaechler) [inline third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java]: broken indent ?

Comment (chibenwa) [reply third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java]: No.

withFileFromClasspath applies for the dockerfiles

withStartupTimeout applies for the container

Spot the nice `))` at the end of line 56

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**